solidity-its/src/main/java/org/sonarsource/solidity/its/SolidityRuling.java

```diff
@@ -56,6 +70,92 @@ public static void collectSolidityFiles() {
       });
   }
 
+  public static void collectFiles() {
+    Arrays.asList(getProjects())
+      .stream()
+      .forEach(projectName -> {
+        List<File> listFiles = new ArrayList<>();
+        try {
+          String path = String.format("%s%s", DIR, projectName);
+          Files.find(Paths.get(path),
+            Integer.MAX_VALUE,
+            (filePath, fileAttr) -> fileAttr.isRegularFile())
+            .filter(file -> file.getFileName().toString().endsWith(".sol"))
+            .forEach(file -> listFiles.add(file.toFile()));
+          sortList(listFiles);
+          filesToAnalyze.put(projectName, listFiles);
+        } catch (IOException e) {
+          LOG.debug(e.getMessage(), e);
+        }
+      });
+  }
+
+  private static void sortList(List<File> listFiles) {
+    Collections.sort(listFiles, new Comparator<File>() {
+      @Override
+      public int compare(File o1, File o2) {
+        String name1 = o1.getName();
+        String name2 = o2.getName();
+        return name1.compareTo(name2);
+      }
+    });
+  }
+
+  public static void collectFilesWithRecordedIssues() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=31&issues=AWRqjT0tMt9UOhrwBY2j&open=AWRqjT0tMt9UOhrwBY2j)

[ISSUE_KEY]: AWRqjT0tMt9UOhrwBY2j